modules/oe_whitelabel_list_pages/tests/src/Functional/ListPagesTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_whitelabel_list_pages_test\FunctionalJavascript;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Drupal\Tests\oe_whitelabel\Functional\WhitelabelBrowserTestBase;
+use Drupal\Tests\sparql_entity_storage\Traits\SparqlConnectionTrait;
+
+/**
+ * Tests the whitelabel oe_list_pages.
+ */
+class ListPagesTest extends WhitelabelBrowserTestBase {
+
+  use SparqlConnectionTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'oe_whitelabel_list_pages_test',
+  ];
+
+  /**
+   * Test fields an oe_whitelabel list page.
+   */
+  public function testListPage(): void {
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+
+    $this->drupalLogin($this->rootUser);
+
+    // Create some test nodes.
+    for ($i = 0; $i < 12; $i++) {
+      $date = new DrupalDateTime('20-10-2020');
+      $values = [
+        'title' => 'News number ' . $i,
+        'type' => 'oe_sc_news',
+        'body' => 'This is content number ' . $i,
+        'status' => NodeInterface::PUBLISHED,
+        'created' => $date->getTimestamp(),
+      ];
+      $node = Node::create($values);
+      $node->save();
+    }
+
+    // Index content.
+    $list_source_factory = $this->container->get('oe_list_pages.list_source.factory');
+    $item_list = $list_source_factory->get('node', 'oe_sc_news');
+    $item_list->getIndex()->indexItems();
+
+    // Create list page for News.
+    $this->drupalGet('/node/add/oe_list_page');
+    $page->fillField('Title', 'News list page');
+
+    // Check that select bundle is required.
+    $page->pressButton('Save');
+    $assert_session->elementTextEquals('css', 'div.alert-content', 'Source bundle field is required.');
+
+    $page->selectFieldOption('Source bundle', 'News');
+    $page->pressButton('Save');
```

Comment:
no need for this, we just want to assert the front functionality and rendering, just create the list page node programatically (I think I left an example of this in the POC, not sure)